### DISCUSSION TITLE
Way to add options later?

### DISCUSSION THREAD

**Discussion by Attackly** [norkunas/youtube-dl-php#182] [Q&A] [answered]
Hey is there an way to add Options later and downloading the video after all the Options have been added?

**Comment by norkunas** [comment]
Hey,
Sorry but I failed to understand, could you be more precise? :wink:

**Reply by Attackly** [reply]
Sure. 

This is the Example code 
```
$collection = $yt->download(
    Options::create()
        ->downloadPath('/path/to/downloads')
        ->url('https://www.youtube.com/watch?v=oDAw7vW7H0c')
);
```

I would like to sometimes add things later on before downloading like the filename. 


so something like this 
```
$collection = $yt->download(
    Options::create()
        ->downloadPath('/path/to/downloads')
        ->url('https://www.youtube.com/watch?v=oDAw7vW7H0c')


//some code 

//adding options later 
$yt->output("Some video");

);
```

I hope its clear what i mean.

**Comment by norkunas** [comment, marked as answer]
You can modify the `Options` instance up to that point when you call `$yt->download()` because this call expands your options and starts the download process. But each `$yt->download()` call can be provided with any options.

**Reply by Attackly** [reply]
Got it Thanks
